### PR TITLE
[Finishes #88063228] Specific error for eval case mismatch

### PR DESCRIPTION
**Pull request by Bjvanminnen**
We want to make the following easier
image
Goal here was "Hello Universe" and i input "Hello universe".

The way I do that is to extract all the strings from the eval object. I only extract those that are the text portion of an EvalText (meaning I ignore things like color, style). I also handle the case where we have an EvalMulti with multiple EvalText objects.

You get only get a mismatch error when you have the same set of strings ignoring case. Thus, if you have a bad string and a case mismatch, you won't get it.  That said, this check does happen before we actually compare the images. This means if you have other mistakes in your image (i.e. you were supposed to rotate and didn't) and a case mismatch, you'll get the case mismatch error. I think this is fine.

### CODE REVIEW

File: apps/src/eval/eval.js [hunk 1]
```diff
@@ -228,6 +233,60 @@ function getDrawableFromBlockXml(blockXml) {
 }
 
 /**
+ * Parse an EvalObject looking for EvalText objects. For each one, extract the
```
Comment (bcjordan) [inline apps/src/eval/eval.js]: Maybe add `Recursively` at beginning of comment

File: apps/src/eval/eval.js [hunk 2]
```diff
@@ -228,6 +233,60 @@ function getDrawableFromBlockXml(blockXml) {
...
+  var strs2 = Eval.getTextStringsFromObject_(object2);
+
+  if (strs1.length !== strs2.length) {
+    return false;
+  }
+
+  strs1.sort();
+  strs2.sort();
+
+  var mismatch = false;
```
Comment (bcjordan) [inline apps/src/eval/eval.js]: maybe `caseMismatch` (could make later part clearer)

File: apps/src/eval/eval.js [hunk 3]
```diff
@@ -228,6 +233,60 @@ function getDrawableFromBlockXml(blockXml) {
...
+
+  for (var i = 0; i < strs1.length; i++) {
+    var str1 = strs1[i];
+    var str2 = strs2[i];
+    if (str1.toLowerCase() !== str2.toLowerCase()) {
```
Comment (bcjordan) [inline apps/src/eval/eval.js]: Had a bit of trouble parsing this section, could alternatively do something like:

```
if (str1 !== str2) {
  if (str1.toLowerCase === str2.toLowerCase) {
    caseMismatch  = true;
  } else {
    return false; // strings differ by more than case
  }
} 
```

**Comment by bcjordan** [comment]
LGTM! Just a couple of tiny nits

File: apps/src/eval/eval.js (hunk 3 above)
Comment (Bjvanminnen) [inline apps/src/eval/eval.js]: Yeah, it felt a little strange to me as I was writing it too. Your way looks better.

**Pull request merged**